lib/application_insights/rack/track_request.rb

```diff
@@ -82,6 +86,23 @@ def format_request_duration(duration_seconds)
 
         Time.at(duration_seconds).gmtime.strftime("00.%H:%M:%S.%7N")
       end
+
+      def request_id_header(env, id)
+        request_id_header = env['HTTP_REQUEST_ID']
+
+        if request_id_header
+          request_id_starts_correctly = (request_id_header =~ /^\|/)
+
+          if request_id_starts_correctly
+            request_id_has_end = (request_id_header =~ /(.|_)$/)
+            request_id_header << '.' unless request_id_has_end
+
+            return "#{request_id_header}#{id}."
```

Comment:
wouldn't this ID be too long? You appending quite a long id to one that was recieved.

Reply:
It looks like there is an 8k header length limit in some browsers, I'm happy to drop the length of the id?

Reply:
perhaps halting it to 16 chars (in line with the .net SDK)

Reply:
I believe in hierarchical mode it is even shorter. Like 4 chars or 6 - I can check out for you